### PR TITLE
bump tonic version to v0.9

### PR DESCRIPTION
**Pull request by xxchan**
... I'm wondering whether we need to bump major version https://github.com/madsim-rs/madsim/issues/145

changelog: https://github.com/hyperium/tonic/blob/master/CHANGELOG.md

I mainly want this: build:: Support boxing fields

### CODE REVIEW

**Comment by wangrunji0408** [comment]
> ... I'm wondering whether we need to bump major version https://github.com/madsim-rs/madsim/issues/145

I guess it's needed. Tonic has breaking API change from 0.8 to 0.9.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Changed
+
+- tonic: Update to `tonic` v0.9.3.
```
Comment (wangrunji0408) [inline CHANGELOG.md:11]:
```suggestion
- tonic: Update to `tonic` v0.9.2.
```

**Comment by xxchan** [comment]
Oh, arrow-flight >= 37 also needs it. That's a larger need.

**Comment by xxchan** [comment]
Although I want to completely understand #153 and  #145  first 😅

**Comment by ZENOTME** [comment]
Also needed in https://github.com/risingwavelabs/risingwave/pull/10875 for introducing Icelake.

**Review by wangrunji0408 — APPROVED** [review]
Approve myself. :)
@xxchan PTAL

**Comment by xxchan** [comment]
I’m ok. Merge first to unblock work. I might need some more time to think
about it, but we can fix problems later since we are 0.x 🤣

**Pull request merged**